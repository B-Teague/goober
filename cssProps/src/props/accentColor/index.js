//Do not modify this file.  This file is generated from ./scripts/generateProps.js
import { default as appendArgs } from '../../util/index'

/**
 * @name accentColor
 * @category CssProps Helper
 * @summary Returns css property accentColor and value as an object
 *
 * @description
 * Returns an object with the css property accentColor with its passed in values
 *
 * @param - One or more parameters that will be appended together based on a delimiter
 * @returns accentColor property as object
 */

export default function accentColor() {
    return {
        accentColor: appendArgs(arguments)
    }
}