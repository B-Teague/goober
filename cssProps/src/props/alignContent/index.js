//Do not modify this file.  This file is generated from ./scripts/generateProps.js
import { default as appendArgs } from '../appendArgs/index'

/**
 * @name alignContent
 * @category CssProps Helper
 * @summary Returns css property alignContent and value as an object
 *
 * @description
 * Returns an object with the css property alignContent with its passed in values
 *
 * @param - One or more parameters that will be appended together based on a delimiter
 * @returns alignContent property as object
 */

export default function alignContent() {
    return {
        alignContent: appendArgs(arguments)
    }
}