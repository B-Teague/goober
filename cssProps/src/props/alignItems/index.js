/* Do not modify this file.  This file is generated from ./cssProps/templates/props.prop.index.mustache */

/**
 * @name alignItems
 * @category CssProps Helper
 *
 * @description
 * Returns an object with the css property alignItems with its passed in values
 *
 * @param {string} One or more parameters that will be appended together separated by a delimiter based on the css syntax
 * @returns alignItems property as object
 */

export default function alignItems(...args) {
    return {
        alignItems: args.join(' ')
    };
}
