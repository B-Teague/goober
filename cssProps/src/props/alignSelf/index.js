/* Do not modify this file.  This file is generated from ./cssProps/templates/props.prop.index.mustache */
import { default as appendArgs } from '../../util/index';

/**
 * @name alignSelf
 * @category CssProps Helper
 *
 * @description
 * Returns an object with the css property alignSelf with its passed in values
 *
 * @param {string} One or more parameters that will be appended together separated by a delimiter based on the css syntax
 * @returns alignSelf property as object
 */

export default function alignSelf() {
    return {
        alignSelf: appendArgs(arguments)
    };
}
