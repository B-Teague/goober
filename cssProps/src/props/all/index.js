//Do not modify this file.  This file is generated from ./scripts/generateProps.js
import { default as appendArgs } from '../appendArgs/index'

/**
 * @name all
 * @category CssProps Helper
 * @summary Returns css property all and value as an object
 *
 * @description
 * Returns an object with the css property all with its passed in values
 *
 * @param - One or more parameters that will be appended together based on a delimiter
 * @returns all property as object
 */

export default function all() {
    return {
        all: appendArgs(arguments)
    }
}