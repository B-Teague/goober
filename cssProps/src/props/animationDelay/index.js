/* Do not modify this file.  This file is generated from ./cssProps/templates/props.prop.index.mustache */

/**
 * @name animationDelay
 * @category CssProps Helper
 *
 * @description
 * Returns an object with the css property animationDelay with its passed in values
 *
 * @param {string} One or more parameters that will be appended together separated by a delimiter based on the css syntax
 * @returns animationDelay property as object
 */

export default function animationDelay(...args) {
    return {
        animationDelay: args.join(' ')
    };
}
