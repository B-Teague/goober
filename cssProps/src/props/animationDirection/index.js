//Do not modify this file.  This file is generated from ./scripts/generateProps.js
import { default as appendArgs } from '../appendArgs/index'

/**
 * @name animationDirection
 * @category CssProps Helper
 * @summary Returns css property animationDirection and value as an object
 *
 * @description
 * Returns an object with the css property animationDirection with its passed in values
 *
 * @param - One or more parameters that will be appended together based on a delimiter
 * @returns animationDirection property as object
 */

export default function animationDirection() {
    return {
        animationDirection: appendArgs(arguments)
    }
}