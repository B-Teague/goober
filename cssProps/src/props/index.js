export { default as display } from './props/display/index'