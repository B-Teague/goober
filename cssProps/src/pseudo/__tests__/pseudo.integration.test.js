import { h, createContext, render } from 'preact';
import { useContext } from 'preact/hooks';
import { setup, extractCss, css, styled } from 'goober';
import { color, background } from '../../props/index';
import { hover } from '../index';

describe('pseudoSelectors', () => {
    it('regular', () => {
        setup(h);

        const target = document.createElement('div');

        const cssClass = css([
            hover([
                color("blue"),
                background("orange")
            ])
        ])

        render(
            <div className={cssClass}>
            </div>,
            target
        );

        expect(extractCss()).toMatchSnapshot();
        expect(target.innerHTML).toMatchSnapshot();
    });

    it('with theme', () => {
        const ThemeContext = createContext();
        const useTheme = () => useContext(ThemeContext);

        setup(h, null, useTheme);

        const target = document.createElement('div');

        const ThemeContainer = styled('div')((props) => ([
            hover([
                color("orange"),
                background(props.theme.color)
            ])
        ]));

        render(
            <ThemeContext.Provider value={{ color: 'blue' }}>
                <ThemeContainer/>
            </ThemeContext.Provider>,
            target
        );

        expect(extractCss()).toMatchSnapshot();
        expect(target.innerHTML).toMatchSnapshot();
    });
});
