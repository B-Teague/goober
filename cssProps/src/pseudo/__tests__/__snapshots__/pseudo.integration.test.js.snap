// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pseudoSelectors regular 1`] = `" .go1382657062:hover{color:blue;background:orange;}"`;

exports[`pseudoSelectors regular 2`] = `"<div class=\\"go1382657062\\"></div>"`;

exports[`pseudoSelectors with theme 1`] = `".go1912523078:hover{color:orange;background:blue;}"`;

exports[`pseudoSelectors with theme 2`] = `"<div class=\\"go1912523078\\"></div>"`;
